Multiple tasks: Windows specific config/script changes and mark room for deletion instead of actually deleting it.

chiragmerchant opened the pull request:
Make file and package.json: updated to work on windows, use test-win instead of test 
Mark for deletion instead of deleting
test suite updated to account for deletedRoom:false filter for rename, fetch, create queries and delete mutation now only updates flag using findOneAndUpdate instead of deleteOne 
Added extra room to data-default.js already marked as deleted

chiragmerchant committed:
Multiple tasks

Make file and package.json: updated to work on windows, use test-win instead of test
Mark for deletion instead of deleting
test suite updated to account for deletedRoom:false filter for rename, fetch, create queries and delete mutation now only updates flag using findOneAndUpdate instead of deleteOne
Added extra room to data-default.js already marked as deleted

chiragmerchant committed:
Fixing failed Prettier checks

Ran the correct prettier extension and enabled, resaved files with issues.

aaronshiel commented on src/schemas/mutation/room-delete.ts:
```diff
@@ -2,21 +2,37 @@
-      _id: args.roomId,
-    });
-    return true;
+    // await RoomModel.deleteOne({
```
remove commented code

aaronshiel commented on src/schemas/mutation/room-delete.ts:
```diff
@@ -2,21 +2,37 @@
-  resolve: async (_root: any, args: { roomId: string }): Promise<boolean> => {
-    const room = await RoomModel.findOne({ _id: args.roomId });
+  resolve: async (_root: any, args: { roomId: string }): Promise<Room> => {
+    console.log(`Looking for ${args.roomId}`)
```
remove log

aaronshiel commented on test/init.spec.ts:
```diff
@@ -28,6 +28,7 @@ after(async () => {
 });
 
 mongoUnit.start().then((url) => {
+  console.log(url)
```
remove log

chiragmerchant committed:
Addressed PR #6 changes + 1 fix

Addressed PR #6 changes requested. (remove console.log and commented code)

Added filter for deleted rooms
Added another test case 'fails if marked for deletion room id'.

aaronshiel reviewed (changes requested):
run `make test-all` and make sure everything passes. I see a lot of warnings for unused variables and "any" types where typing can be used.

Some help:
`_root: any,` can be changed to `_root: GraphQLObjectType,`

chiragmerchant committed:
Makefile and package.json run script entries, formatting, license etc

Makefile and npm now suitable for windows development, checks and tests
Added license text to license and header
Updated licenses in all code, ran format

Addresses requests from PR 6
Still pending 5 warnings.

chiragmerchant committed:
Disabling warnings for now

disable  no-unused-vars for 5 warnings (3 lines)

chiragmerchant committed:
Remove unused code, commented code, eslint disables

Remove variables corsOptions CORS_ORIGIN
